spec/lib/hubspot/company_properties_spec.rb

```diff
@@ -14,158 +14,325 @@
+
+          response = Hubspot::CompanyProperties.create!(create_params)
+
+          assert_hubspot_api_request(
+            :post,
+            "/companies/v2/properties",
+            body: {
+              name: name,
+              groupName: "companyinformation",
+              options: []
+            }
+          )
+          expect(response["name"]).to eq(name)
+
+          Hubspot::CompanyProperties.delete!(name)
         end
       end
 
-      context 'with all valid parameters' do
-        cassette 'company_properties/create_property'
+      context "with no valid parameters" do
+        it "returns nil" do
+          response = Hubspot::CompanyProperties.create!({})
 
-        it 'should return the valid parameters' do
-          response = Hubspot::CompanyProperties.create!(params)
-          valid_params.each { |k, v| expect(response[k]).to eq(v) }
+          expect(response).to be nil
         end
       end
     end
 
-    describe '.update!' do
-      context 'with no valid parameters' do
-
-        it 'should return nil ' do
-          expect(Hubspot::CompanyProperties.update!(params['name'], {})).to be(nil)
+    describe ".update!" do
+      it "updates the company property" do
+        VCR.use_cassette "company_properties/update_property" do
+          name = "fax_number"
+          label = "fax number label"
+          new_label = "new fax number label"
+
+          Hubspot::CompanyProperties.create!({
+            groupName: "companyinformation",
+            label: label,
+            name: name,
+            type: "string",
+          })
+
+          update_params = {
+            groupName: "companyinformation",
+            label: new_label,
+            type: "string",
+          }
+
+          response = Hubspot::CompanyProperties.update!(name, update_params)
+
+          assert_hubspot_api_request(
+            :put,
+            "/companies/v2/properties",
+            body: {
+              groupName: "companyinformation",
+              label: new_label,
+              options: [],
+              type: "string",
+            }
+          )
+          expect(response["label"]).to eq(new_label)
+
+          Hubspot::CompanyProperties.delete!(name)
         end
       end
 
-      context 'with mixed parameters' do
-        cassette 'company_properties/update_property'
+      context "with no valid parameters" do
+        it "returns nil" do
+          company_property_name = "fax_number"
+          params = {}
 
-        it 'should return the valid parameters' do
-          params['description']       = 'What is their favorite flavor?'
-          valid_params['description'] = params['description']
+          response = Hubspot::CompanyProperties.update!(
+            company_property_name,
+            params
+          )
 
-          response = Hubspot::CompanyProperties.update!(params['name'], params)
-          valid_params.each { |k, v| expect(response[k]).to eq(v) }
+          expect(response).to be nil
```

Comment:
Same here, is it worth expecting the status code to be something?

Reply:
The `update!` function returns early with `nil` when given empty params. An API call is not issued.